onnxruntime/core/framework/session_state_initializer.h

```diff
@@ -23,21 +24,20 @@ class Logger;
 // Don't use this class before graph partition is done
 class SessionStateInitializer {
  public:
-  SessionStateInitializer(onnxruntime::Graph& graph,
-                          SessionState& session_state,
-                          const ExecutionProviders& providers,
+  SessionStateInitializer(const std::basic_string<PATH_CHAR_TYPE>& graph_loc, onnxruntime::Graph& graph,
+                          SessionState& session_state, const ExecutionProviders& providers,
                           KernelRegistryManager& kernel_registry_manager);
 
   // First perform any transformations and create the execution plan
   common::Status CreatePlan(const std::vector<NodeArg*>& outer_scope_node_args,
                             bool enable_sequential_execution);
 
   // initialize tensors, and save. save kernels and input/output node mappings
-  // @param enable_memory_pattern
-  common::Status InitializeAndSave(bool enable_memory_pattern,
-                                   const std::vector<NodeArg*>* implicit_inputs = nullptr);
+  // \param implicit_inputs could be NULL
+  common::Status InitializeAndSave(const std::vector<NodeArg*>* implicit_inputs);
 
  private:
+  const std::basic_string<PATH_CHAR_TYPE>& graph_loc_;
   onnxruntime::Graph& graph_;
```

Comment:
May want to take a copy of the string. I don't think the original string is guaranteed to stay around for the duration of the session like the other references here.

Reply:
SessionStateInitializer is only a short-lived temporal object. 

---
In reply to: [261378681](https://github.com/Microsoft/onnxruntime/pull/520#discussion_r261378681) [](ancestors = 261378681)